lib/matplotlib/axes/_axes.py

```diff
@@ -8547,7 +8569,7 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
             fillcolor = linecolor = self._get_lines.get_next_color()
 
         # Check whether we are rendering vertically or horizontally
-        if vert:
+        if orientation == 'vertical':
```

Comment:
With this, you ignore the `vert` parameter of existing code during the deprecation.

You in the conditional vert handling block above, you have to reassign `orientation` based on `vert` if `vert` is not None. Since we additionally do not have a not-set information on `orientation` and it's not worth to introduce that, the rule for the deprecation period must be: If `vert` is given, it overrides any `orientation` setting.